spec/src/main/asciidoc/core/spi.asciidoc

```diff
@@ -1580,3 +1594,35 @@ CDI 2.0 introduced the following Configurators interface:
 * <<observer_method_configurator>> for `ObserverMethod` configuration
 
 The container must provide implementation for all these configurators and make them available in matching container lifecycle events as defined in <<init_events>>.
+
+[[interception_proxy_factory]]
+
+=== Apply interceptor and decorators programmatically
+
+Contextual instances of bean having interceptors binding or defined decorators are provided by the container as a proxy.
+
+When a bean is defined by a producer method or has a custom bean type, the container doesn't provide this proxy for interceptor or decorator.
+
+To allow creation of these proxies programmatically, CDI 2.0 introduces the `javax.enterprise.inject.spi.InterceptionProxyFactory<T>` interface where T is the type of instance to create the proxy for.
+
+[source, java]
+----
+public interface InterceptionProxyFactory<T> {
+    InterceptionProxyFactory<T> ignoreFinalMethods();
+    AnnotatedTypeConfigurator<T> configure();
+    <U extends T> U createInterceptionProxy(T instance);
+}
+----
+
+* `ignoreFinalMethods()` forces the creation of the proxy even if the type T contains public final methods.
+This method will bypass rules defined in <<unproxyable>>
+Final methods on T won't be available on the produced proxy.
+* `configure()` returns an `AnnotatedTypeConfigurator` (as defined in <<annotated_type_configurator>>) initialized with the instance type to easily apply specific interceptor binding to apply when creating the proxy.
+The method always return the same `AnnotatedTypeConfigurator`
+* `createInterceptionProxy()` returns the proxy adding interceptors and decorators on the provided instance.
+The method can be only called once, subsequent calls will throw an exception.
+
+An `InterceptionProxyFactory` can be obtain be calling `BeanManager.createInterceptionFactory` as defined in <<bm_obtain_interception_proxy_factory>>
+
+The container must also provide an instance of `InterceptionProxyFactory` for every producer methods injecting this interface.
```

Comment:
Must provide a built-in bean with bean type `javax.enterprise.inject.spi.InterceptionProxyFactory`, qualifier `@Default` and _"injectable only in a producer method parameter"_ (needs rewording).